External/Plugins/CodeRefactor/Provider/RenamingHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using CodeRefactor.Commands;
+using PluginCore.FRService;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using ASCompletion.Completion;
+using ASCompletion.Context;
+using ASCompletion.Model;
+using PluginCore.Localization;
+using PluginCore.Managers;
+using ProjectManager.Helpers;
+
+namespace CodeRefactor.Provider
+{
+    class RenamingHelper
+    {
+        static readonly List<Rename> queue = new List<Rename>();
+        static Rename currentCommand;
+
+        public static void AddToQueue(ASResult target)
+        {
+            AddToQueue(target, true);
+        }
+        public static void AddToQueue(ASResult target, bool outputResults)
+        {
+            string name = RefactoringHelper.GetRefactorTargetName(target);
+            string label = TextHelper.GetString("Label.NewName");
+            string title = string.Format(TextHelper.GetString("Title.RenameDialog"), name);
+            LineEntryDialog askName = new LineEntryDialog(title, label, name);
+            if (askName.ShowDialog() == DialogResult.OK && askName.Line.Trim().Length > 0 && askName.Line.Trim() != name)
+            {
+                string newName = askName.Line.Trim();
+                queue.Add(new Rename(target, outputResults, newName));
+                if (ASContext.Context.CurrentModel.haXe && target.Member != null &&
+                    (target.Member.Flags & (FlagType.Getter | FlagType.Setter)) > 0)
+                {
+                    List<MemberModel> list = target.Member.Parameters;
+                    if (list[0].Name == "get") RenameMember(target.InClass, $"get_{name}", $"get_{newName}", outputResults);
```

Comment:
String interpolation requires C# 6 / VS15, I think @elsassph wanted to avoid that for now?